accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -410,24 +417,24 @@ read.
     is treated as a first read and the initializer expression is evaluated
     again.
 
-A write to a field or variable which is marked `final` and `late` is a runtime
+A write to a field or variable which is marked `late` and `final` is a runtime
 error unless the field or variable was declared with no initializer expression,
 and there have been no previous writes to the field or variable (including via
 an initializing formal or an initializer list entry).
 
-Overriding a field which is marked both `final` and `late` with a member which
+Overriding a field which is marked both `late` and `final` with a member which
 does not otherwise introduce a setter introduces an implicit setter which
 throws.  For example:
```

Comment:
That sounds spurious, and wrong. A final late field has a getter and a setter. Those are the ones you override.

Overriding the getter with another getter is valid, so it should also be valid to declare a non-late final field to override that getter.

Otherwise *all* final fields should shadow inherited setters with throwing versions. That's not a `late` property.

Reply:
> A final late field has a getter and a setter.

A final late field _with_ an initializer wouldn't have a setter.

But we could use the standard approach: Getters and setters are overridden by other getters/setters iff the latter exist.

So a late final field with an initializer has a getter and no setter, and the getter may be overridden just like other getters. A setter may be added to the interface by a subtype&mdash;no special exceptions needed.

For a late final field without an initializer there is a getter and a setter, and they can be overridden as usual.

The rule specified in line 418 differs from this by specifying that if a subclass overrides the getter and would inherit the setter then we auto-generate a setter that throws. We could do that, but I agree with @lrhn that we might as well avoid introducing a special exception for this case.

Reply:
Note that my proposed change is a completely presentational matter: The grammar insists that variables have `late final` rather than `final late`, so I adjusted a couple of locations to use the same ordering.

So the discussion about the underlying rule is separate, and I'd prefer to take that discussion in a separate PR.